fastly.admin.inc

```diff
@@ -98,6 +98,75 @@ function fastly_setup_form($form_state) {
   return system_settings_form($form);
 }
 
+/**
+ * Webhook form
+ *
+ * @param $form_state
+ * @return mixed
+ */
+function fastly_webhook_form($form_state) {
+  $form['fastly_webhook_enabled'] = array(
+    '#type' => 'radios',
+    '#title' => t('Webhook enabled?'),
+    '#options' => array(
+        'yes' => 'Yes',
+        'no' => 'No',
+    ),
+    '#default_value' => variable_get('fastly_webhook_enabled', 'no'),
+  );
+
+  $defaultWebhookUrlValue = variable_get('fastly_webhook_url', '');
+
+  $form['fastly_webhook_url'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Incoming WebHook URL'),
+    '#description' => t('Incoming WebHook URL'),
+    '#states' => array(
+        'visible' => array(
+            ':input[name="fastly_webhook_enabled"]' => array('value' => 'yes'),
+        ),
+    ),
+    '#required' => TRUE,
+    '#default_value' => $defaultWebhookUrlValue,
+  );
+
+  $defaultFastlyWebhookEvents = variable_get('fastly_webhook_events', array());
+
+  $form['fastly_webhook_events'] = array(
+    '#type' => 'select',
+    '#multiple' => TRUE,
+    '#title' => t('Send webhook notifications for'),
+    '#required' => FALSE,
+    '#options' => array("purge_actions" =>"Purge actions", "vcl_upload_actions" =>"VLC upload actions"),
+    '#size' => 5,
+    '#weight' => 0,
+    '#states' => array(
+        'visible' => array(
+            ':input[name="fastly_webhook_enabled"]' => array('value' => 'yes'),
+        ),
+    ),
+    '#default_value' => $defaultFastlyWebhookEvents,
+  );
+
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save webhook configuration'),
+  );
+
+  return $form;
+}
+
+/**
+ * Save form variables
+ *
+ * @param $form
+ * @param $form_state
+ */
+function fastly_webhook_form_submit(&$form, &$form_state) {
+  variable_set('fastly_webhook_url', $form_state['values']['fastly_webhook_url']);
```

Comment:
Should have an entry in `fastly.install` to remove these variables on uninstall. See: https://github.com/vvuksan/fastly-drupal/blob/7.x-1.x/fastly.install#L11

Reply:
Thanks Grant, I'll push this change today during the day.